dandiapi/api/tests/test_embargo.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+
+from dandiapi.api.models import Dandiset
+
+
+@pytest.fixture(
+    params=[
+        Dandiset.EmbargoStatus.EMBARGOED,
+        Dandiset.EmbargoStatus.UNEMBARGOING,
+    ]
+)
+def embargo_status(request):
+    return request.param
+
+
+EMPTY_PAGINATION = {
+    'count': 0,
+    'next': None,
+    'previous': None,
+    'results': [],
+}
+
+
+@pytest.mark.parametrize(
+    ('method', 'url_format'),
+    [
+        ('get', '/api/dandisets/{dandiset.identifier}/versions/'),
+        ('get', '/api/dandisets/{dandiset.identifier}/versions/draft/'),
+        ('put', '/api/dandisets/{dandiset.identifier}/versions/draft/'),
+        ('delete', '/api/dandisets/{dandiset.identifier}/versions/draft/'),
+        ('get', '/api/dandisets/{dandiset.identifier}/versions/draft/info/'),
+        ('post', '/api/dandisets/{dandiset.identifier}/versions/draft/publish/'),
+        ('get', '/api/dandisets/{dandiset.identifier}/versions/draft/assets/'),
+        ('post', '/api/dandisets/{dandiset.identifier}/versions/draft/assets/'),
+        ('get', '/api/dandisets/{dandiset.identifier}/versions/draft/assets/paths/'),
+        ('get', '/api/dandisets/{dandiset.identifier}/versions/draft/assets/{asset.asset_id}/'),
+        ('put', '/api/dandisets/{dandiset.identifier}/versions/draft/assets/{asset.asset_id}/'),
+        ('delete', '/api/dandisets/{dandiset.identifier}/versions/draft/assets/{asset.asset_id}/'),
+        (
+            'get',
+            '/api/dandisets/{dandiset.identifier}/versions/draft/assets/{asset.asset_id}/download/',
+        ),
+        (
+            'get',
+            '/api/dandisets/{dandiset.identifier}/versions/draft/assets/{asset.asset_id}/info/',
+        ),
+        (
+            'get',
+            '/api/dandisets/{dandiset.identifier}/versions/draft/assets/{asset.asset_id}/validation/',  # noqa: E501
+        ),
+    ],
+)
+@pytest.mark.django_db
+def test_embargo_visibility(
+    api_client,
+    user,
+    dandiset_factory,
+    draft_version_factory,
+    draft_asset_factory,
+    embargoed_asset_blob,
+    embargo_status,
+    method,
+    url_format,
+):
+    dandiset = dandiset_factory(embargo_status=embargo_status)
+    version = draft_version_factory(dandiset=dandiset)
+    asset = draft_asset_factory(blob=None, embargoed_blob=embargoed_asset_blob)
+    version.assets.add(asset)
+
+    url = url_format.format(dandiset=dandiset, asset=asset)
+    response = getattr(api_client, method)(url)
+    # The client is not authenticated, so all response codes should be 401
+    assert response.status_code == 401
+
+    api_client.force_authenticate(user=user)
+    response = getattr(api_client, method)(url)
+    # The client is now authenticated but not an owner, so all response codes should be 403
+    assert response.status_code == 403
```

Comment:
Is successful access tested elsewhere?

Reply:
Searching `test.*embargo` gives the full catalog of embargo tests. Upload to embargoed dandisets is thoroughly tested. Other operations are tested more intermittently, but that should be sufficient since the `get_queryset()` functionality should extend to every operation in the ViewSet.

It would be nice to keep other embargo tests here, but I think it makes more sense to keep them next to their unembargoed counterparts, since a lot of logic is duplicated.